\documentclass[aspectratio=169,12pt,t]{beamer}
\usepackage{graphicx}
\setbeameroption{hide notes}
\setbeamertemplate{note page}[plain]
\usepackage{listings}
\usepackage{eepic}

\input{header.tex}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% end of header
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% title info
\title{A data mishap}
\subtitle{Allele frequencies in sibships}
\author{\href{https://kbroman.org}{Karl Broman}}
\institute{Biostatistics \& Medical Informatics, UW{\textendash}Madison}
\date{\href{https://kbroman.org}{\tt \scriptsize \color{foreground} kbroman.org}
\\[-4pt]
\href{https://github.com/kbroman}{\tt \scriptsize \color{foreground} github.com/kbroman}
\\[-4pt]
\href{https://twitter.com/kwbroman}{\tt \scriptsize \color{foreground} @kwbroman}
\\[-4pt]
{\scriptsize Slides: \href{https://kbroman.org/Talk_DataMishap}{\tt kbroman.org/Talk\_DataMishap}}
}


\begin{document}

% title slide
{
\setbeamertemplate{footline}{} % no page number here
\frame{
  \titlepage

\note{
}

} }


\begin{frame}{GWAS for ``{\hilit morning person}''}

  \bigskip

\figh{Figs/GWAS_morning_person.jpeg}{0.75}

\bigskip

\hfill \scriptsize {\lolit Hu et al (2016)
  \href{https://doi.org/10.1038/ncomms10448}{doi:10.1038/ncomms10448}}

\note{
}

\end{frame}




\begin{frame}{BRCA pedigree}

  \bigskip

\figh{Figs/brca_family.png}{0.7}

  \bigskip

\hfill \scriptsize {\lolit Hall et al (1990)
  \href{https://doi.org/10.1126/science.2270482}{doi:10.1126/science.2270482}}

\note{
}

\end{frame}





\begin{frame}[c]{Affected sib pairs}
\only<1|handout 0>{\figw{Figs/sibpairs.pdf}{1.0}}
\only<2>{\figw{Figs/sibpairs_wdata.pdf}{1.0}}

\note{
  In the late 1990s, one of ways we tried to identify disease genes
  was with affected sibling pair studies. You gather a bunch of pairs
  of siblings who were both affected with a disease, and then get
  genotype data for them, and look for genomic regions where the
  affected sibpairs had more similar genotypes than you would expect
  by chance.
}

\end{frame}


\begin{frame}[c]{IBS vs IBD}


\large

IBS = identical by {\hilit state}

{\color{background} IBS}    = same allele number \\[18pt]

  IBD = identical by {\hilit descent}

{\color{background} IBD} = copies of the same ancestral allele \\[36pt]

non-inbred sibs are IBD = 0, 1, 2

{\color{background} non-inbred sibs} with probability = 1/4, 1/2, 1/4



\note{
   In measuring genetic similarity at a locus, it's valuable to
   distinguish between alleles being ``identical by state'' (meaning
   they just look the same) and ``identical by descent'' (meaning that
   they are copies of an ancestral allele). Non-inbred siblings will
   have IBD status 0, 1, or 2, with probability 1/4, 1/2, 1/4,
   respectively.
}

\end{frame}

\begin{frame}{Prostate cancer genome scan}

\bigskip

\figh{Figs/gh_results_bad.pdf}{0.75}

\note{
  In an affected sib-pair study, we'll scan across the genome; at each
  point will seek to estimate the proportion of alleles shared IBD
  between affected siblings and compare that to what is expected for
  siblings.

  The first such study I was involved in was of prostate cancer, and
  consisted of maybe 150 affected sibling pairs. This plot (of
  $-\log_{10}$ p-values) is an approximation of my initial results.
  We're looking for values around 3, so these were super exciting to
  me. I distinctly remember faxing these results to my collaborators,
  thinking ``I am so awesome. I will conquer all diseases.''
}

\end{frame}


\begin{frame}{}

  \vspace*{-0.7in} \hspace*{-1.0in}

\figw{Figs/moustache.jpeg}{1.14}

\vspace*{-1.10in}
\hfill
{\large \color{white} so happy} \hspace*{0.75in}

\end{frame}


\begin{frame}{}

\bigskip \bigskip

\figh{Figs/fax_machine.png}{0.7}

\bigskip \bigskip

\hfill \scriptsize \href{https://bit.ly/faxpic}{\tt \lolit bit.ly/faxpic}

\end{frame}



\begin{frame}[c]{Lesson}

\centering
\Large
If it seems too good to be true, \\[12pt]
it probably is.


\note{
  But as soon as I sent that fax, I was like, ``Huh. Those results
  seem too good to be true.''

  It turns out that I'd messed up the allele frequencies and so the
  results were all messed up.
}

\end{frame}



\begin{frame}[c]{Prostate cancer pairs}
\only<1|handout 0>{\figw{Figs/sibpairs_nopar.pdf}{1.0}}
\only<2>{\figw{Figs/sibpairs_nopar_wdata.pdf}{1.0}}

\note{
   In this prostate cancer study, the affected sibpairs are all old,
   and there's essentially no data on the parents. In this case,
   determining the number of alleles shared IBD turns out to be
   particularly sensitive to the allele frequencies.

   For example, if they're both 3/3 and 3 is relatively rare, that
   it's very likely that they're IBD=2. But if 3 is quite common, then
   they could reasonably be IBD=1 or 0.

   If they both share a 6 allele, is that IBD=1, or IBD=0? If 6 is
   rare, you'd lean towards IBD=1, but if 6 is common, it could be
   either.
}

\end{frame}



\begin{frame}[c]{Prostate cancer genome scan -- corrected}
\figw{Figs/gh_results_good.pdf}{1.0}

\note{
   The unusually strong results I got were entirely due to a mistake
   in the code that estimated the allele frequencies. If I use more
   reasonable estimates, this is what I get. There's maybe evidence
   for a disease locus on chr 16 and possibly also 15, but the
   evidence isn't very strong.

   And this is sort of what we'd expect given the size of this study.
   We're hoping to find some evidence of a disease gene, but we're not
   going to see the whole genome lighting up.
}

\end{frame}





\begin{frame}[c]{}

\bigskip \bigskip

\fighboxed{Figs/broman2001.png}{0.8}

\onslide<2>{
\vspace*{-2in} \hspace*{0.5in}
\fighboxed{Figs/broman2001_erratum.png}{0.65}
}

\hfill \scriptsize \href{https://doi.org/10.1002/gepi.2}{\tt \lolit doi:10.1002/gepi.2}

\end{frame}






\begin{frame}[c]{}

\Large

Slides: \href{https://kbroman.org/Talk_DataMishap}{\tt kbroman.org/Talk\_DataMishap}

\vspace{7mm}

\href{https://kbroman.org}{\tt \lolit kbroman.org}

\vspace{7mm}

\href{https://github.com/kbroman}{\tt \lolit github.com/kbroman}

\vspace{7mm}

\href{https://twitter.com/kwbroman}{\tt \lolit @kwbroman}


\end{frame}




\end{document}
