\documentclass[aspectratio=169,12pt,t]{beamer}
\usepackage{graphicx}
\setbeameroption{hide notes}
\setbeamertemplate{note page}[plain]
\usepackage{listings}
\usepackage{eepic}

\input{header.tex}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% end of header
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% title info
\title{allele frequencies in sibships}
\subtitle{a data mishap}
\author{\href{https://kbroman.org}{Karl Broman}}
\institute{Biostatistics \& Medical Informatics, UW{\textendash}Madison}
\date{\href{https://kbroman.org}{\tt \scriptsize \color{foreground} kbroman.org}
\\[-4pt]
\href{https://github.com/kbroman}{\tt \scriptsize \color{foreground} github.com/kbroman}
\\[-4pt]
\href{https://twitter.com/kwbroman}{\tt \scriptsize \color{foreground} @kwbroman}
\\[-4pt]
{\scriptsize Slides: \href{https://kbroman.org/Talk_DataMishap}{\tt kbroman.org/Talk\_DataMishap}}
}


\begin{document}

% title slide
{
\setbeamertemplate{footline}{} % no page number here
\frame{
  \titlepage

\note{
These are slides for a 5-min talk about a data mishap, for a community
night (https://www.littlemissdata.com/fdf/datamishapsnight) organized
by Caitlin Hudon (@beeonaposy) and Laura Ellis (@LittleMissData).
}

} }


\begin{frame}{GWAS for ``{\hilit morning person}''}

  \bigskip

\figh{Figs/GWAS_morning_person.jpeg}{0.75}

\bigskip

\hfill \scriptsize {\lolit Hu et al (2016)
  \href{https://doi.org/10.1038/ncomms10448}{doi:10.1038/ncomms10448}}

\note{
   Genome-wide association studies (GWAS) have been a revolution
   in human genetics. This figure is from a study of 23andMe
   participants who were asked whether they're a morning person. This
   binary trait was associated with genotype at markers across the
   genome, immediately showing genes associated with the trait.
}

\end{frame}




\begin{frame}{BRCA pedigree}

  \bigskip

\figh{Figs/brca_family.png}{0.7}

  \bigskip

\hfill \scriptsize {\lolit Hall et al (1990)
  \href{https://doi.org/10.1126/science.2270482}{doi:10.1126/science.2270482}}

\note{
  Back in the day, gene discovery involved the collection and analysis
  of large families. This is one of the families from the study that
  identified the BRCA1 gene. An important insight there was focusing
  on families with early-onset breast cancer.
}

\end{frame}





\begin{frame}[c]{Affected sib pairs}
\only<1|handout 0>{\figw{Figs/sibpairs.pdf}{1.0}}
\only<2>{\figw{Figs/sibpairs_wdata.pdf}{1.0}}

\note{
  In-between, there was a period where we thought we could find
  disease genes by gathering a moderate number of affected sibling
  pairs.  You look for regions where affected sibpairs had more
  similar genotypes than you would expect by chance.
}

\end{frame}


\begin{frame}[c]{Marshfield, Wisconsin}

  \figh{Figs/marshfield_map.png}{0.9}

\note{
  In 1998 I was a postdoc in a genetics lab in Marshfield, WI (2 1/2
  hours drive north of Madison). My advisor hooked me up with an
  affected sibpair study on prostate cancer. I did the initial data
  cleaning and a basic analysis, hoping to wow the famous people
  involved with my prowess.
}

\end{frame}



\begin{frame}{Prostate cancer genome scan}

\bigskip

\figh{Figs/gh_results_bad.pdf}{0.75}

\note{
  This plot (of $-\log_{10}$ p-values) is an approximation of my
  initial results. We're looking for values around 3, so these were
  super exciting to me: much higher association than I would have
  expected, and on many more chromosomes than I would have expected.
}

\end{frame}


\begin{frame}{}

  \vspace*{-0.7in} \hspace*{-1.0in}

\figw{Figs/moustache.jpeg}{1.14}

\vspace*{-1.10in}
\hfill
{\large \color{white} so happy} \hspace*{0.75in}

\note{
  It was so awesome.
}

\end{frame}


\begin{frame}{}

\bigskip \bigskip

\figh{Figs/fax_machine.png}{0.7}

\bigskip \bigskip

\hfill \scriptsize \href{https://bit.ly/faxpic}{\tt \lolit bit.ly/faxpic}

\note{
  I immediately faxed my results off to my collaborators.
  (That's how we shared results with each other in 1998.)
}

\end{frame}



\begin{frame}[c]{}

\centering
\Large
If it seems too good to be true, \\[12pt]
it probably is.


\note{
  But as soon as I sent that fax, I was like, ``Huh. Those results
  seem too good to be true.''

  It turns out that I'd messed up the allele frequencies and so the
  results were all messed up.
}

\end{frame}



\begin{frame}[c]{Prostate cancer pairs}
\only<1|handout 0>{\figw{Figs/sibpairs_nopar.pdf}{1.0}}
\only<2>{\figw{Figs/sibpairs_nopar_wdata.pdf}{1.0}}

\note{
   In this prostate cancer study, the affected sibpairs are all old,
   and there's essentially no data on the parents. In this case,
   our method for determining sharing is particularly sensitive to the
   allele frequencies.

   It's not obvious how to estimate the allele frequencies, but also
   the simple approach I took had a bug that really through things
   off.
}

\end{frame}



\begin{frame}[c]{Prostate cancer genome scan -- corrected}
\figw{Figs/gh_results_good.pdf}{1.0}

\note{
   The unusually strong results I got were entirely due to a mistake
   in the code that estimated the allele frequencies. If I use more
   reasonable estimates, this is what I get. There's maybe evidence
   for a disease locus on chr 16 and possibly also 15, but the
   evidence isn't very strong.

   And this is sort of what we'd expect given the size of this study.
   We're hoping to find some evidence of a disease gene, but we're not
   going to see the whole genome lighting up.
}

\end{frame}





\begin{frame}[c]{}

\bigskip \bigskip

\fighboxed{Figs/broman2001.png}{0.8}

\onslide<2|handout 0>{
\vspace*{-2in} \hspace*{0.5in}
\fighboxed{Figs/broman2001_erratum.png}{0.65}
}

\hfill \scriptsize \href{https://doi.org/10.1002/gepi.2}{\tt \lolit doi:10.1002/gepi.2}

\note{
  My collaborators were pretty nice about it. And I ended up writing a
  paper about the problem. That paper also had a major flaw, which is
  also interesting and instructive, but that's another story.
}

\end{frame}






\begin{frame}[c]{}

\Large

Slides: \href{https://kbroman.org/Talk_DataMishap}{\tt kbroman.org/Talk\_DataMishap}

\vspace{7mm}

\href{https://kbroman.org}{\tt \lolit kbroman.org}

\vspace{7mm}

\href{https://github.com/kbroman}{\tt \lolit github.com/kbroman}

\vspace{7mm}

\href{https://twitter.com/kwbroman}{\tt \lolit @kwbroman}


\end{frame}




\end{document}
